js/service.js

```diff
@@ -29,13 +29,14 @@ function doXHR(route, params = {}) {
   return new Promise((resolve, reject) => {
     request.open(`GET`, `${route}${combinedParams ? pojoToQuery(combinedParams) : ``}`, true);
 
+    request.withCredentials = true;
     request.onload = (event) => {
       let result = event.currentTarget;
 
       if (result.status >= 200 && result.status < 400) {
-        resolve(JSON.parse(result.response));
+        resolve(result.response);
```

Comment:
this is actually the right place to do the JSON parsing: either result.response is proper JSON in which case we send the parsed data on, or it's not and in that case this XHR handling code is responsible for saying what should happen on a failure because the data it received does not match the data it expected.